libr/core/cmd_anal.c

```diff
@@ -2841,21 +2841,33 @@ static int cmd_anal_fcn(RCore *core, const char *input) {
 		break;
 	case 'd': // "afd"
 		{
+		bool json = false;
 		ut64 addr = 0;
 		if (input[2] == '?') {
 			eprintf ("afd [offset]\n");
 		} else if (input[2] == ' ') {
 			addr = r_num_math (core->num, input + 2);
 		} else {
 			addr = core->offset;
+			if (input[2] == 'j') {
+				json = true;
+			}
 		}
 		RAnalFunction *fcn = r_anal_get_fcn_in (core->anal, addr, 0);
 		if (fcn) {
+			if (json) {
+				r_cons_printf ("{\"name\":\"");
```

Comment:
use the pj api. fcn->name can contain special chars that will break the json. the pj api takes care of this

Reply:
thanks!